src/main/java/ch/njol/skript/events/EvtLeash.java

```diff
@@ -0,0 +1,112 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityUnleashEvent;
+import org.bukkit.event.entity.PlayerLeashEntityEvent;
+import org.bukkit.event.player.PlayerUnleashEntityEvent;
+import org.jetbrains.annotations.Nullable;
+
+@SuppressWarnings({"unchecked", "NotNullFieldNotInitialized"})
+public class EvtLeash extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Leash / Unleash", EvtLeash.class, CollectionUtils.array(PlayerLeashEntityEvent.class, EntityUnleashEvent.class), "[:player] [:un]leash[ing] [[of] %-entitydatas%]")
```

Comment:
I just want to double check that `PlayerUnleashEntityEvent` isn't listed here but I'm assuming it's covered as a subclass of `EntityUnleashEvent` and so doesn't need to be?

Reply:
From my previous testing, EntityUnleashEvent also calls for Player. I could verify this again and get back to you with the result.